activestorage/app/models/active_storage/blob.rb

```diff
@@ -270,7 +270,8 @@ def analyzed?
   # deleted as well or you will essentially have a dead reference. It's recommended to use the +#purge+ and +#purge_later+
   # methods in most circumstances.
   def delete
-    service.delete key
+    service.delete(key)
+    service.delete_prefixed("variants/#{key}/") if image?
```

Comment:
Why do we only want this for `image?` variants? Isn't it a little sneaky that the `delete` behavior is different depending on the blob type?

Reply:
Non-image blobs don’t have variants. Skipping this saves an unnecessary API call.

Reply:
Ah, right 👍